subprojects/luajit/meson.build

```diff
@@ -7,21 +7,23 @@ system_deps = [
     cc.find_library('m', required: false)
 ]
 
-add_project_arguments('-DLUAJIT_ENABLE_LUA52COMPAT', language: 'c')
+# compat flag is needed for both the buildvm code generator (compiled natively) and luajit itself
+add_project_arguments('-DLUAJIT_ENABLE_LUA52COMPAT', language: 'c', native: true)
+add_project_arguments('-DLUAJIT_ENABLE_LUA52COMPAT', language: 'c', native: false)
 if host_machine.system() == 'linux'
-    add_project_arguments('-DLUAJIT_OS=LUAJIT_OS_LINUX', language: 'c')
+    add_project_arguments('-DLUAJIT_OS=LUAJIT_OS_LINUX', language: 'c', native: true)
     readline_dep = cc.find_library('readline')
     ljvm_mode = 'elfasm'
     ljvm_bout = 'lj_vm.s'
 elif host_machine.system() == 'darwin'
-    add_project_arguments(['-DLUAJIT_OS=LUAJIT_OS_OSX'], language: 'c')
+    add_project_arguments(['-DLUAJIT_OS=LUAJIT_OS_OSX'], language: 'c', native: true)
     readline_dep = cc.find_library('readline')
     ljvm_mode = 'machasm'
     ljvm_bout = 'lj_vm.s'
 elif host_machine.system() == 'windows'
-    add_project_arguments('-DLUAJIT_OS=LUAJIT_OS_WINDOWS', language: 'c')
+    add_project_arguments('-DLUAJIT_OS=LUAJIT_OS_WINDOWS', language: 'c', native: true)
     if cc.get_id() != 'msvc'
-        add_project_arguments('-malign-double', language: 'c')
+        add_project_arguments('-malign-double', language: 'c', native: true)
```

Comment:
We don't want this set on cross builds? Really?

Reply:
At least the LuaJIT Makefile only sets it specifically on the build machine and nowhere else: https://github.com/LuaJIT/LuaJIT/blob/v2.1/src/Makefile#L356 (note: HOST_SYS = build_machine, TARGET_SYS = host_machine)

Reply:
Weird. We can leave it then; I'm sure if we try to cross compile we'll run into a host of other issues anyway.